catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/pocketmusic/PocketMusicActivityTest.java

```diff
@@ -160,41 +136,38 @@ public void toggleRandomNoteViews() {
 		for (int i = 0; i < recyclerViewItemCount; i++) {
 			randomPosition[i] = random.nextInt(TrackRowView.QUARTER_COUNT * TrackView.ROW_COUNT);
 			onView(withId(R.id.tact_scroller))
-					.perform(actionOnItemAtPosition(i, clickNoteViewAtPosition(randomPosition[i])));
-			onView(withRecyclerView(R.id.tact_scroller).atPosition(i))
+					.perform(actionOnItemAtPosition(i, toggleNoteViewAtPositionInTact(randomPosition[i])));
+			onView(withTactScroller(R.id.tact_scroller).atPosition(i))
 					.check(matches(withToggledNoteView(randomPosition[i])));
 		}
 
-		pressBack();
-		onView(withText(R.string.pocketmusic_recorded_filename)).perform(click());
+		relaunchActivityOpenJustSavedFile();
 
 		for (int i = 0; i < recyclerViewItemCount; i++) {
 			onView(withId(R.id.tact_scroller)).perform(scrollToPosition(i));
-			onView(withRecyclerView(R.id.tact_scroller).atPosition(i))
+			onView(withTactScroller(R.id.tact_scroller).atPosition(i))
 					.check(matches(withToggledNoteView(randomPosition[i])));
 		}
 	}
 
-	@Test
-	public void playButtonElementExists() {
-		navigateToPocketMusicActivity();
-		onView(withId(R.id.pocketmusic_play_button)).check(matches(isDisplayed()));
-	}
+	private void relaunchActivityOpenJustSavedFile() {
+		pocketMusicActivityRule.getActivity().finish();
 
-	@Test
-	public void testOrientationChange() {
-		navigateToPocketMusicActivity();
-		Activity currentActivity = getCurrentActivity();
-		currentActivity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE);
-		currentActivity.setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
-	}
+		List<SoundInfo> soundInfo = ProjectManager.getInstance().getCurrentSprite().getSoundList();
 
-	@After
-	public void tearDown() throws Exception {
+		Intent pocketMusicDataIntent = null;
+		if (soundInfo != null && !soundInfo.isEmpty()) {
```

Comment:
i guess the test calling this will fail if this is null, but for ease of debugging if something breaks here wouldnt it be nicer to do something like:
assertNotNull(soundInfo)
assertFalse(soundInfo.isEmpty());
instead of the if(...)